eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/OperationSourceProvider.cs

```diff
@@ -0,0 +1,102 @@
+            _clientField = new FieldProvider(FieldModifiers.Private | FieldModifiers.ReadOnly, typeof(ArmClient), "_client", this);
+            _operationSourceInterface = new CSharpType(typeof(IOperationSource<>), _resource.Type);
+        }
+
+        protected override string BuildName() => $"{_resourceName}OperationSource";
+
+        protected override string BuildRelativeFilePath() => Path.Combine("src", "Generated", "LongRunningOperation", $"{Name}.cs");
+
+        protected override CSharpType[] BuildImplements()
+        {
+            return [new CSharpType(typeof(IOperationSource<>), _resource.Type)];
+        }
+
+        protected override MethodProvider[] BuildMethods() => [BuildCreateResult(), BuildCreateResultAsync()];
+
+        private MethodProvider BuildCreateResultAsync()
+        {
+            var signature = new MethodSignature(
+                "CreateResultAsync",
+                null,
+                MethodSignatureModifiers.Async,
+                new CSharpType(typeof(ValueTask<>), _resource.Type),
+                $"",
+                [KnownAzureParameters.Response, _cancellationTokenParameter],
+                ExplicitInterface: _operationSourceInterface);
+            var body = new MethodBodyStatement[]
+            {
+                UsingDeclare("document", typeof(JsonDocument), Static(typeof(JsonDocument)).Invoke(nameof(JsonDocument.ParseAsync), [KnownAzureParameters.Response.Property(nameof(Response.ContentStream)), Default, _cancellationTokenParameter], true), out var documentVariable),
+                Declare("data", _resourceData.Type, Static(_resourceData.Type).Invoke($"Deserialize{_resourceData.Name}", documentVariable.Property(nameof(JsonDocument.RootElement)), New.Instance<ModelReaderWriterOptions>(Literal("W"))), out var dataVariable),
+                Return(New.Instance(_resource.Type, [_clientField, dataVariable])),
+            };
+            return new MethodProvider(signature, body, this);
+        }
+
+        private MethodProvider BuildCreateResult()
+        {
+            var signature = new MethodSignature(
+                "CreateResult",
+                null,
+                MethodSignatureModifiers.None,
+                _resource.Type,
+                $"",
+                [KnownAzureParameters.Response, _cancellationTokenParameter],
+                ExplicitInterface: _operationSourceInterface);
+            var body = new MethodBodyStatement[]
+            {
+                UsingDeclare("document", typeof(JsonDocument), Static(typeof(JsonDocument)).Invoke(nameof(JsonDocument.Parse), [KnownAzureParameters.Response.Property(nameof(Response.ContentStream))]), out var documentVariable),
+                Declare("data", _resourceData.Type, Static(_resourceData.Type).Invoke($"Deserialize{_resourceData.Name}", documentVariable.Property(nameof(JsonDocument.RootElement)), New.Instance<ModelReaderWriterOptions>(Literal("W"))), out var dataVariable),
```

Comment:
`new ModelReaderWriterOptions("W")` is needed here because of options is not optional
https://github.com/Azure/azure-sdk-for-net/blob/44a1cc564d07a8f964f29a77581e8849861b62d2/eng/packages/http-client-csharp/generator/TestProjects/Local/Mgmt-TypeSpec/src/Generated/FooData.Serialization.cs#L69